### PR TITLE
Allow custom cellRenderer views

### PR DESCRIPTION
**Pull request by mschindlerMM**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,13 +9,22 @@
...
   "license": "MIT",
   "peerDependencies": {
     "ag-grid-community": "^26.2.1",
     "ag-grid-enterprise": "^26.2.1"
+  },
+  "devDependencies": {
+    "@parcel/transformer-elm": "^2.3.2",
+    "@webcomponents/custom-elements": "^1.5.0",
+    "ag-grid-community": "^26.2.1",
+    "ag-grid-enterprise": "^26.2.1",
```
Comment (Tranquility) [inline package.json:26]: Are these dependencies for the example app? I thought we don't want to have them as dependencies in here...

File: README.md [hunk 2]
```diff
@@ -66,3 +66,77 @@ To support browsers that don't have built-in support for the Web Components API,
 import "@webcomponents/custom-elements";
 import "@mercurymedia/elm-ag-grid";
 ```
+
+## Custom views for cells
+
+Custom cell content (such as a button or link) can be created and used as a separate Elm application. This application must be defined as `Browser.element` in load it into the table cell.
```
Comment (Tranquility) [inline README.md:72]: Not as a separate elm application, but a separate elm app can be used as a renderer for cells.

File: README.md [hunk 3]
```diff
@@ -66,3 +66,77 @@ To support browsers that don't have built-in support for the Web Components API,
...
+
+### Utilizing a component in Elm
+
+To use a component for cell rendering, you can use the `AppRenderer { componentName : String, componentParams : Maybe Json.Encode.Value } (dataType -> String)` renderer in your column definition.
+
+```elm
+    { field = "details"
+    , renderer = AppRenderer { componentName = "linkRenderer", componentParams = Nothing } .url
+    , headerName = "Details"
+    , settings = { defaultSettings | editable = False }
```
Comment (Tranquility) [inline README.md:126]: Should we enforce the `editable = False` by always setting it to this when the `AppRenderer` is used?

File: ag-grid-webcomponent/app_renderer.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+class AppRenderer {
+  /**
+   * Gets called once before the renderer is used.
+   * Loads the Elm application into a `DocumentFragment`.
+   * @param {*} params Contains encoded user and table data
+   */
+  init(params) {
+    this.eGui = new DocumentFragment();
+
+    const component = window.ElmAgGridComponentRegistry[params.componentName];
```
Comment (Tranquility) [inline ag-grid-webcomponent/app_renderer.js:15]: We could show a nice error message that says that the app is not added to the component registry if we can not find it.

File: ag-grid-webcomponent/index.js [hunk 5]
```diff
@@ -29,7 +30,7 @@ class AgGrid extends HTMLElement {
   }
 
   loadAttribute(attr) {
-    let gridElement = document.querySelector("ag-grid").getAttribute(attr);
+    let gridElement = this.getAttribute(attr);
```
Comment (Tranquility) [inline ag-grid-webcomponent/index.js:33]: Would have been a bit nicer to have this in a separate commit because it is not really related to the renderer stuff. But I am fine with keeping it like this.

**Review by Tranquility — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (mschindlerMM) [reply README.md]: I changed the sentences a bit. Should sound more accurate now.

File: README.md (hunk 3 above)
Comment (mschindlerMM) [reply README.md:126]: No. You might still want to be able to edit the cell value normally. Even though you could still handle the editing in the component, I don't think we should enforce that.

Comment (Tranquility) [reply README.md:126]: Right.

File: package.json (hunk 1 above)
Comment (mschindlerMM) [reply package.json:26]: These are just devDependencies. So they will not be required when using the package. It's just for local development and running the example app.

**Review by Tranquility — APPROVED** [review]
